src/_posts/security/bulletins/2024-03-13-SSB-2024-002.md

```diff
@@ -0,0 +1,106 @@
+---
+title: SSB-2024-002 - Password Hash Algorithm Update For PostgreSQL Users
+nav:  SSB-2024-002 (MD5)
+modified_at: 2024-03-13 00:00:00
+tags: security bulletin scalingo postgresql md5 scram-sha-256 database
+---
+
+This bulletin announces a minor update in our database authentication systems. Effective from $MIGRATION_START_DATE, Scalingo will be migrating the password hashing algorithm for existing PostgreSQL databases from MD5 to SCRAM (Salted Challenge Response Authentication Mechanism).
+
+PostgreSQL databases created after the 20th of December 2023 **already use SCRAM-SHA-256** as the default authentication method.
+
+The MD5 hashing algorithm was the *de facto* standard when we started offering Database-As-A-Service in 2014. Nowadays it is considered insecure, because in the unlikely event a database breach, an attacker could use the hashes to infer the original passwords (using a technique called *[rainbow tables](https://en.wikipedia.org/wiki/Rainbow_table)*).
+
+As part of our ongoing efforts to enhance security and provide the best possible service to our customers, **we will update the password hashing algorithm  from `MD5` to `SCRAM-SHA-256`.**
+
+Quote from the PostgreSQL documentation:
+
+> The method scram-sha-256 performs SCRAM-SHA-256 authentication, as described in RFC 7677. It is a challenge-response scheme that prevents password sniffing on untrusted connections and supports storing passwords on the server in a cryptographically hashed form that is thought to be secure.
+> 
+> 
+> This is the most secure of the currently provided methods, but it is not supported by older client libraries.
+
+This is a per-database-user change. It implies resetting the users' passwords (since PostgreSQL doesn't know them).
+
+## Timeline of the rollout
+
+There are 2 distinct timelines, depending on the type of the parent application of the PostgreSQL add-on.
+
+- **Health Data Hosting databases (HDH):** for PostgreSQL add-ons linked to HDH applications, this update is scheduled for $MIGRATION_START_DATE+6_MONTHS (in 6 months' time). This notice is due to the nature of the data contained in these databases and the excessive caution they require.
+- **Other databases:** this change will be applied automatically **in the coming weeks**.
+
+You will be notified by e-mail 24 hours before the start of the maintenance period of the affected add-on(s), according to the database maintenance window configuration.
+
+## Impact on your applications
+
+Rest assured, this transition is designed to have minimal impact on your applications (less than a second). The SCRAM-SHA-256 algorithm is widely supported by PostgreSQL clients library and has been in use for several years.
```

Comment:
Are all the current sessions closed? Is the application to be restarted?

Reply:
Connections will not close, no restart needed.